cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -412,38 +390,81 @@ def test_cloud_provision_from_template_with_attached_disks(
     if provider.one_of(AzureProvider):
         recursive_update(inst_args, {'environment': {'availability_zone': provisioning("av_set")}})
 
-    device_name = "/dev/sd{}"
+    device_name = "vd{}"
     device_mapping = []
 
-    with provider.mgmt.with_volumes(1, n=disks) as volumes:
-        for i, volume in enumerate(volumes):
-            device_mapping.append((volume, device_name.format(chr(ord("b") + i))))
-        # Set up automate
+    volumes = provider.mgmt.volume_configurations(1, n=disks)
+
+    @request.addfinalizer
+    def delete_volumes():
+        for volume in volumes:
+            provider.mgmt.delete_volume(volume)
+
+    # Set up automate
+    for i, volume in enumerate(volumes, 0):
+        if i == 0:
+            provider.mgmt.capi.volumes.set_bootable(volume, True)
+            device_mapping.append(
```

Comment:
Consider only setting bootable in this if, and using the same dictionary instead of repeating.

```
device_mapping.append(dict(
    boot_index=0 if i==0 else -1,
    uuid=volume,
    device_name=device_name.format(chr(ord('a') + i)))
)
~``

Reply:
Well there is that `provider.mgmt.capi.volumes.set_bootable(volume, True)` that needs to be set only if `i == 0` so I need double ifs then. But whatever. I don't think this is a big deal.